Improvements in Victory.py

MeuhV1 opened the pull request:
Removed double function
Discover first x era technology goals fail if target is not met and all techs from that era are discovered
Discover first x era technology goals only checks when the discovered era is the era for the UHV
Tamil trade UHV stores decimal values
Fixed Enter first era goals

dguenms commented on Assets/Python/Victory.py:
```diff
@@ -2811,6 +2811,16 @@ def countFirstDiscovered(iPlayer, iEra):
 			iCount += 1
 	return iCount
 	
+def isFirstDiscoveredPossible(iPlayer, iEra, iRequired):
```
Wouldn't be better to just count techs first discovered by `iPlayer` and all undiscovered techs `countFirstDiscovered(-1, iEra)` and see if they add up to at least `iRequired`?

MeuhV1 replied:
Yeah. I'll change it.

MeuhV1 commented:
Updated

dguenms commented on Assets/Python/Victory.py:
```diff
@@ -1599,18 +1599,24 @@ def onTechAcquired(iPlayer, iTech):
 			if countFirstDiscovered(iPlayer, iRenaissance) >= 8 and countFirstDiscovered(iPlayer, iIndustrial) >= 8:
 				if iPlayer == iEngland: win(iEngland, 2)
 				else: lose(iEngland, 2)
+			if not isFirstDiscoveredPossible(iEngland iRenaissance, 8) or not isFirstDiscoveredPossible(iEngland, iIndustrial, 8):
```
Looks like there is a comma missing here for all these invocations.

MeuhV1 commented:
Updated.

I also included an additional condition so the UHV checks only if the discovered tech belongs to the era of the tech goals for speed improvement.

MeuhV1 commented:
Rebased to latest DoC
Included Tamil UHV update
Fixed Enter first era goals

dguenms commented on Assets/Python/Victory.py:
```diff
@@ -532,21 +532,21 @@ def checkTurn(iGameTurn, iPlayer):
 		# third goal: acquire 4000 gold by trade by 1200 AD
 		if isPossible(iTamils, 2):
-			iTradeGold = 0
+			iTradeGold = 0.0
```
Instead of a float, this should still store an integer but instead by a factor of 100, Civ4 does not need more accuracy.

MeuhV1 replied:
Only iTradeGold of all instances of fTamilTradeGold?

dguenms replied:
All instances where floats are used to store the accrued Tamil gold.

MeuhV1 replied:
Done.

I also included Italy to the list of conditional spawns and moved the code inside the check of line 1129.

Commit:
Removed double function
Discover first x era technology goals fail if target is not met and all techs from that era are discovered
Discover first x era technology goals only checks when the discovered era is the era for the UHV
Tamil trade UHV stores decimal values
Fixed Enter first era goals
Added Italy to the list of conditional spawns

MeuhV1 commented:
Used int * 100 for Tamil UHV

I also included Italy to the list of conditional spawns and moved the code inside the check of line 1129 of RNF.py
